tlspuffin/src/put_registry.rs

```diff
@@ -1,33 +1,46 @@
-use puffin::{
-    put::PutName,
-    put_registry::{Factory, PutRegistry},
-};
+use puffin::{put::PutName, put_registry::PutRegistry};
+#[cfg(feature = "cputs")]
+use tls_harness::C_PUT_TYPE;
 
 use crate::protocol::TLSProtocolBehavior;
 
 pub const OPENSSL111_PUT: PutName = PutName(['O', 'P', 'E', 'N', 'S', 'S', 'L', '1', '1', '1']);
 pub const WOLFSSL520_PUT: PutName = PutName(['W', 'O', 'L', 'F', 'S', 'S', 'L', '5', '2', '0']);
 pub const TCP_PUT: PutName = PutName(['T', 'C', 'P', '_', '_', '_', '_', '_', '_', '_']);
 
-pub const TLS_PUT_REGISTRY: PutRegistry<TLSProtocolBehavior> = PutRegistry {
-    factories: &[
-        crate::tcp::new_tcp_factory,
-        #[cfg(feature = "openssl-binding")]
-        crate::openssl::new_openssl_factory,
-        #[cfg(feature = "wolfssl-binding")]
-        crate::wolfssl::new_wolfssl_factory,
-    ],
-    default: DEFAULT_PUT_FACTORY,
-};
-
-pub const DEFAULT_PUT_FACTORY: fn() -> Box<dyn Factory<TLSProtocolBehavior>> = {
-    cfg_if::cfg_if! {
-        if #[cfg(feature = "openssl-binding")] {
-            crate::openssl::new_openssl_factory
-        } else if #[cfg(feature = "wolfssl-binding")] {
-            crate::wolfssl::new_wolfssl_factory
-        } else {
-             crate::tcp::new_tcp_factory
+pub fn tls_registry() -> PutRegistry<TLSProtocolBehavior> {
+    #[cfg(feature = "cputs")]
+    extern "C" fn callback(put: *const C_PUT_TYPE) {
+        if put.is_null() {
+            return;
         }
+
+        println!("registering C put")
```

Comment:
Is this a debug statement?

Reply:
I guess it is not, but should we maybe use a logger here?

Reply:
Right now, it is a placeholder until we include the first real C PUT. I decided to use `println!` so that I'm sure to see this callback being called, regardless of the log level. This only appears when the `cputs` feature is set anyway, so it should not be a disturbance for the current applications.